### ISSUE TITLE
"translation" namespace no longer works after upgrading to i18next v21.1.1

### ISSUE THREAD

**Issue by fromi** [i18next/react-i18next#1384]
## 🐛 Bug Report

I just upgraded to i18next v21.1.1.
This code, which used to work, no longer works:
```
const {t} = useTranslation(['specificNS', 'translation'])
[...]
t('translation|Some text')
```
Using i18next v20.6.1, the text is translated just fine. The issue comes from upgrading i18next.
The fact that 'translation' is the default namespace seems to take part in causing the bug: in some other files of my project, I have:
```
const {t} = useTranslation(['translation', 'specificNS'])
[...]
t('Some text')
t('specificNS|Some specific text')
```
And it works just fine there.

## To Reproduce

Unfortunately, I was unable to reproduce the issue inside a Code Sandbox.
I do not understand what is causing this side effect, however it is real, and it is undocumented in the i18next v21 changelog.
The project uses next.js.

**Comment by jamuhl** [comment]
If you can't reproduce it on codesandbox we won't be able to confirm / fix the potential issue

**Comment by fromi** [comment]
Yes, well I'm available at anytime for a screen sharing session to confirm the issue if you like.
Otherwise, I will stick with i18next v20 for some time.

**Comment by jamuhl** [comment]
Are you able to produce a failing test? https://github.com/i18next/react-i18next/blob/master/test/useTranslation.spec.js#L55

**Comment by adrai** [comment]
fyi: you mentioned next.js so it may be it is because of this line: https://github.com/isaachinman/next-i18next/blob/d43be9b8055e222c34eb88a886d6b55cc488cdd8/src/config/createConfig.ts#L147

So you can try to set defaultNS to "translation" in your option, that should fix it.

**Comment by fromi** [comment]
I already tried to set defaultNS to "translation" (among many other things to try to corner the issue), but it does not help unfortunately.
I did many thing in CodePen to try to reproduce the issue without success also, so I don't feel like I could produce a failing test either without digging a lot into the framework, which I cannot afford unfortunately :/
Anyway, I will keep version 20 for now. Maybe someone else will face the same regression but with a better understanding of what is causing the issue.
